runtime/src/identity.rs

```diff
@@ -380,6 +396,136 @@ decl_module! {
             Ok(())
         }
 
+        /// Initiates master key rotation for a DID.
+        ///
+        /// # Failure
+        /// Only called by master key owner.
+        ///
+        /// # Arguments
+        /// * `did` caller's DID
+        /// * `new_key` master key replacing current key
+        pub fn change_master_key(origin, did: IdentityId, new_key: Key) -> Result {
```

Comment:
We could allow the caller to specify an expiry for the proposed new key to accept rather than assuming None.